packages/block-editor/src/components/media-placeholder/index.js

```diff
@@ -52,84 +51,74 @@ const InsertFromURLPopover = ( { src, onChange, onSubmit, onClose } ) => (
 	</URLPopover>
 );
 
-export class MediaPlaceholder extends Component {
-	constructor() {
-		super( ...arguments );
-		this.state = {
-			src: '',
-			isURLInputVisible: false,
-		};
-		this.onChangeSrc = this.onChangeSrc.bind( this );
-		this.onSubmitSrc = this.onSubmitSrc.bind( this );
-		this.onUpload = this.onUpload.bind( this );
-		this.onFilesUpload = this.onFilesUpload.bind( this );
-		this.openURLInput = this.openURLInput.bind( this );
-		this.closeURLInput = this.closeURLInput.bind( this );
-	}
-
-	onlyAllowsImages() {
-		const { allowedTypes } = this.props;
-		if ( ! allowedTypes ) {
-			return false;
-		}
-		return every( allowedTypes, ( allowedType ) => {
-			return (
-				allowedType === 'image' || startsWith( allowedType, 'image/' )
-			);
-		} );
-	}
-
-	componentDidMount() {
-		this.setState( { src: get( this.props.value, [ 'src' ], '' ) } );
-	}
+export function MediaPlaceholder( {
+	value = {},
+	allowedTypes = [],
+	className,
+	icon,
+	labels = {},
+	mediaPreview,
+	notices,
+	isAppender,
+	accept,
+	addToGallery,
+	multiple = false,
+	dropZoneUIOnly,
+	disableDropZone,
+	disableMediaButtons,
+	onError,
+	onSelect,
+	onCancel,
+	onSelectURL,
+	onDoubleClick,
+	onFilesPreUpload = noop,
+	onHTMLDrop = noop,
+	children,
+} ) {
+	const mediaUpload = useSelect( ( select ) => {
+		const { getSettings } = select( 'core/block-editor' );
+		return getSettings().mediaUpload;
+	}, [] );
+	const [ src, setSrc ] = useState( '' );
+	const [ isURLInputVisible, setIsURLInputVisible ] = useState( false );
+
+	useEffect( () => {
+		setSrc( value?.src ?? '' );
+	}, [ value ] );
+
+	const onlyAllowsImages = () =>
+		allowedTypes?.every(
```

Comment:
This optional chaining shouldn't be necessary since `allowedTypes` defaults to `[]`.
```suggestion
		allowedTypes.every(
```